cmd/certificate_maker/certificate_maker.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2024 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// Package main implements a certificate creation utility for Fulcio.
+// It supports creating root and leaf certificates using (AWS, GCP, Azure).
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/sigstore/fulcio/pkg/certmaker"
+	"github.com/sigstore/fulcio/pkg/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	"go.uber.org/zap"
+)
+
+// CLI flags and env vars for config.
+// Supports AWS KMS, Google Cloud KMS, and Azure Key Vault configurations.
+var (
+	version string
+
+	rootCmd = &cobra.Command{
+		Use:     "fulcio-certificate-maker",
+		Short:   "Create certificate chains for Fulcio",
+		Long:    `A tool for creating root, intermediate, and leaf certificates for Fulcio with code signing capabilities`,
+		Version: version,
+	}
+
+	createCmd = &cobra.Command{
+		Use:   "create",
+		Short: "Create certificate chain",
+		RunE:  runCreate,
+	}
+
+	kmsType              string
+	kmsRegion            string
+	kmsKeyID             string
+	kmsTenantID          string
+	kmsCredsFile         string
+	rootTemplatePath     string
+	leafTemplatePath     string
+	rootKeyID            string
+	leafKeyID            string
+	rootCertPath         string
+	leafCertPath         string
+	intermediateKeyID    string
+	intermediateTemplate string
+	intermediateCert     string
+	kmsVaultToken        string
+	kmsVaultAddr         string
+)
+
+func mustBindPFlag(key string, flag *pflag.Flag) {
+	if err := viper.BindPFlag(key, flag); err != nil {
+		log.Logger.Fatal("failed to bind flag", zap.String("flag", key), zap.Error(err))
+	}
+}
+
+func mustBindEnv(key, envVar string) {
+	if err := viper.BindEnv(key, envVar); err != nil {
+		log.Logger.Fatal("failed to bind env var", zap.String("var", envVar), zap.Error(err))
+	}
+}
+
+func init() {
+	log.ConfigureLogger("prod")
+
+	rootCmd.AddCommand(createCmd)
+
+	createCmd.Flags().StringVar(&kmsType, "kms-type", "", "KMS provider type (awskms, gcpkms, azurekms, hashivault)")
+	createCmd.Flags().StringVar(&kmsRegion, "aws-region", "", "AWS KMS region")
+	createCmd.Flags().StringVar(&kmsKeyID, "kms-key-id", "", "KMS key identifier")
+	createCmd.Flags().StringVar(&kmsTenantID, "azure-tenant-id", "", "Azure KMS tenant ID")
```

Comment:
same here and below, can the var names include the provider if these aren't meant to be generalized across KMS providers?